presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/json/JsonRowEncoder.java

```diff
@@ -0,0 +1,358 @@
+        }
+    }
+
+    private String formatDateTime(EncoderColumnHandle columnHandle, SqlTimeWithTimeZone value)
+    {
+        switch (columnHandle.getDataFormat()) {
+            case "custom-date-time":
+                return this.getCustomFormatter(columnHandle.getFormatHint())
+                        .print(org.joda.time.Instant.ofEpochMilli(value.getMillisUtc()).toDateTime());
+            case "rfc2822":
+                return Instant.ofEpochMilli(value.getMillisUtc())
+                        .atZone(UTC)
+                        .format(RFC_FORMATTER);
+            case "iso8601":
+                return Instant.ofEpochMilli(value.getMillisUtc())
+                        .atZone(UTC)
+                        .toLocalDateTime()
+                        .format(ISO_DATE_TIME);
+            case "milliseconds-since-epoch":
+                return String.valueOf(value.getMillisUtc());
+            case "seconds-since-epoch":
+                return String.valueOf(MILLISECONDS.toSeconds(value.getMillisUtc()));
+            default:
+                throw new PrestoException(GENERIC_USER_ERROR, format("invalid data format '%s' defined for column '%s'", columnHandle.getDataFormat(), columnHandle.getName()));
+        }
+    }
+
+    private String formatDateTime(EncoderColumnHandle columnHandle, SqlTimestamp value)
+    {
+        switch (columnHandle.getDataFormat()) {
+            case "custom-date-time":
+                return this.getCustomFormatter(columnHandle.getFormatHint())
+                        .print(org.joda.time.Instant.ofEpochMilli(value.getMillis()).toDateTime());
+            case "rfc2822":
+                return Instant.ofEpochMilli(value.getMillis())
+                        .atZone(UTC)
+                        .format(RFC_FORMATTER);
+            case "iso8601":
+                return Instant.ofEpochMilli(value.getMillis())
+                        .atZone(UTC)
+                        .toLocalDateTime()
+                        .format(ISO_DATE_TIME);
+            case "milliseconds-since-epoch":
+                return String.valueOf(value.getMillis());
+            case "seconds-since-epoch":
+                return String.valueOf(MILLISECONDS.toSeconds(value.getMillis()));
+            default:
+                throw new PrestoException(GENERIC_USER_ERROR, format("invalid data format '%s' defined for column '%s'", columnHandle.getDataFormat(), columnHandle.getName()));
+        }
+    }
+
+    private String formatDateTime(EncoderColumnHandle columnHandle, SqlTimestampWithTimeZone value)
+    {
+        switch (columnHandle.getDataFormat()) {
+            case "custom-date-time":
+                return this.getCustomFormatter(columnHandle.getFormatHint())
+                        .print(org.joda.time.Instant.ofEpochMilli(value.getMillisUtc()).toDateTime());
+            case "rfc2822":
+                return Instant.ofEpochMilli(value.getMillisUtc())
+                        .atZone(UTC)
```

Comment:
I think that we should preserve the timezone when it's available, instead of expressing as UTC timestamps.